lib/monitor.cpp

```diff
@@ -0,0 +1,173 @@
+#include <atomic>
+#include <thread>
+#include <list>
+#include <sstream>
+#include <fstream>
+
+#include "monitor.h"
+#include "device.h"
+#include "quda_api.h"
+#include "util_quda.h"
+#include "tune_quda.h" // hash, version, resource path
+
+namespace quda
+{
+
+  namespace monitor
+  {
+
+    static std::thread monitor_thread;
+    static std::atomic<int> check(0);
```

Comment:
Some comments on what these two variables are might be useful. I guess `check` could use some more intuitive name like `is_running`?

Reply:
[3f7939e](https://github.com/lattice/quda/pull/1479/commits/3f7939e18e29334c59c7fb9d753549658c2b9104)